test/Reflex/Test/CrossImpl.hs

```diff
@@ -102,16 +106,19 @@ testAgreement :: (Eq c, Eq d, Show c, Show d) => (forall m t. TestCaseConstraint
 testAgreement builder inputs = do
   let identityResult = testPure builder inputs
   tracePerf "---------" $ return ()
-  let spiderResult = testSpider builder inputs
+  let spiderResult = unsafePerformIO $ S.runSpiderHost $ testTimeline builder inputs
   tracePerf "---------" $ return ()
-  let resultsAgree = identityResult == spiderResult
+  let profiledResult = unsafePerformIO $ S.runSpiderHost $ Prof.runProfiledM $ testTimeline builder inputs
```

Comment:
For someone unfamiliar with this code like myself, I would need a nice comment explaining why `unsafePerformIO` is ok here.

Reply:
Especially considering you have access to IO here?

Reply:
Yeah it shouldn't be necessary here.